### DISCUSSION TITLE
how to get the original file from chunks in minio

### DISCUSSION THREAD

**Discussion by starry-eyed-art** [juicedata/juicefs#2252] [Q&A]
We use minio for storage to avoid the problem of data loss. At the same time, in order to meet the scenario of writing data directly to disk in the business, we use juiceFs to connect to minio. The question now is, how can we restore the files originally placed in the disk directory according to the chunk on the minio.

我们使用了minio做存储来避免数据丢失的问题，同时为了满足业务中直接往磁盘上写数据的场景，我们使用了juiceFs来对接minio。现在的问题是，我们如何根据minio上面的chunk来恢复原来放置在磁盘目录中的文件。

**Comment by xiaogaozi** [comment]
Hi, you can't restore files directly from objects in MinIO, you have to go through the metadata in the JuiceFS metadata engine.

**Reply by starry-eyed-art** [reply]
Hi, I tried load and dump, and found that I can only restore the data to the metadata engine through metadata, but I still can't get the original specific file.

hi，我尝试了load和dump，发现通过metadata我只能把数据恢复到metadata engine之中，但是还是无法获得原来的具体的文件。

**Reply by xiaogaozi** [reply]
What is the specific meaning of "get the original specific file"?

**Reply by starry-eyed-art** [reply]
比如我使用juicefs做磁盘文件到minio的中间代理，每当我写入一个文件到磁盘，那么juiceFs就会将这个文件切割成多个并上传到minio。这里的磁盘上的文件就是上述”原来的具体的文件“。
我这边想要的是能否通过juicefs metadata将minio的文件同步到我另一个机器的磁盘上。
我尝试过sync命令，但是sync恢复的是minio中已经被切割后的文件，并不是我原来的文件。

**Reply by xiaogaozi** [reply]
Is your requirement to copy files from JuiceFS to another machine? Or need to access files in JuiceFS on another machine?

If your requirement is the former, mount the JuiceFS file system on another machine, and then copy it to the local disk via `juicefs sync` or `cp` command.

If it is the latter, you can also mount the JuiceFS file system on another machine.
